src/classes/RD2_ApiService.cls

```diff
@@ -0,0 +1,178 @@
+                    // but not paused, then mark it as not current
+                    schedule.isCurrent = false;
+                }
+                schedules.add(schedule);
+            }
+
+            schedulesByRdId.put(rd.Id, schedules);
+        }
+
+        return JSON.serialize(schedulesByRdId);
+    }
+
+    /**
+     * @return SOQL of Recurring Donation records with a subquery on schedule records
+     */
+    private String getRecurringDonationQuery() {
+        Set<String> queryFields = new Set<String>{
+            'Id',
+            'EndDate__c'
+        };
+        if (UserInfo.isMultiCurrencyOrganization()) {
+            queryFields.add('CurrencyIsoCode');
+        }
+
+        queryFields.add(getScheduleSubQuery());
+
+        String soql = new UTIL_Query()
+            .withFrom(npe03__Recurring_Donation__c.SObjectType)
+            .withSelectFields(queryFields)
+            .withWhere('Id IN :rdIds')
+            .build();
+
+        return soql;
+    }
+
+    /**
+     * @description Returns query on Recurring Donation Schedule
+     * Return all schedules for the RD Schedule visualization.
+     * The Schedule Service will filter them out if not applicable within the time period.
+     * @return String RD Schedule subquery used in the Recurring Donation SOQL
+     */
+    public String getScheduleSubQuery() {
+        Set<String> fields = new RD2_ScheduleService.ScheduleQueryHandler().getFields();
+        fields.add('CreatedDate');
+
+        return '(SELECT ' + String.join(new List<String>(fields), ', ') +
+            ' FROM RecurringDonationSchedules__r' +
+            ' ORDER By StartDate__c' +
+            ')';
+    }
+
+    /***
+    * @description Wrapper class for Recurring Donation Schedule data for the querySchedules api
+    */
+    public class Schedule {
+
+        public String recurringDonationId { get; set; }
+        public String scheduleId { get; set; }
+        public Boolean isActive { get; set; }
+        public Boolean isCurrent { get; set; }
+        public Boolean isPause { get; set; }
+        public String campaignId { get; set; }
+        public String dayOfMonth { get; set; }
+        public Date endDate { get; set; }
+        public Decimal installmentAmount { get; set; }
+        public Integer installmentFrequency { get; set; }
+        public String installmentPeriod { get; set; }
+        public String paymentMethod { get; set; }
+        public Date startDate { get; set; }
+        public String statusReason { get; set; }
+        public String currencyCode { get; set; }
+        public Datetime createdDate { get ;set; }
+
+        public Schedule(RecurringDonationSchedule__c rdSchedule, npe03__Recurring_Donation__c rd) {
+            isActive = rdSchedule.Active__c;
+            isCurrent = rdSchedule.StartDate__c <= RD2_ScheduleService.currentDate
+                && (rdSchedule.EndDate__c == null || rdSchedule.EndDate__c >= RD2_ScheduleService.currentDate);
```

Comment:
Do you please mind using "<=" instead, ie `|| RD2_ScheduleService.currentDate <= rdSchedule.EndDate__c`?